SimPPS/PPSPixelDigiProducer/interface/RPixChargeShare.h

```diff
@@ -0,0 +1,32 @@
+#ifndef SimPPS_PPSPixelDigiProducer_RPix_Charge_Share_H
+#define SimPPS_PPSPixelDigiProducer_RPix_Charge_Share_H
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "Geometry/VeryForwardGeometry/interface/CTPPSPixelSimTopology.h"
+#include "SimPPS/PPSPixelDigiProducer/interface/RPixSignalPoint.h"
+
+class RPixChargeShare {
+public:
+  RPixChargeShare(const edm::ParameterSet &params, uint32_t det_id);
+  std::map<unsigned short, double> Share(const std::vector<RPixSignalPoint> &charge_map);
+
+private:
+  uint32_t det_id_;
+  std::vector<double> signalCoupling_;
+  CTPPSPixelSimTopology theRPixDetTopology_;
+  CTPPSPixelIndices pxlInd;
+  const int pxlRowSize_ = pxlInd.getDefaultRowDetSize();
+  const int pxlColSize_ = pxlInd.getDefaultColDetSize();
+  int no_of_pixels_;
+
+  double sqrt_2;
```

Comment:
@mundim , is it const? or it is a variable, if const not necessary to be class member.

Reply:
done.